src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -2104,15 +2100,49 @@ class BeginEndSegment(BaseSegment):
         Ref("DelimiterSegment", optional=True),
         Indent,
         AnyNumberOf(
-            OneOf(
-                Ref("BeginEndSegment"),
+            Ref("StatementSegment"),
+            min_times=1,
+        ),
+        Dedent,
+        "END",
+    )
+
+
+@tsql_dialect.segment()
+class TryCatchSegment(BaseSegment):
+    """A `TRY/CATCH` block pair.
+
+    https://docs.microsoft.com/en-us/sql/t-sql/language-elements/try-catch-transact-sql?view=sql-server-ver15
+    """
+
+    type = "try_catch"
+    match_grammar = Sequence(
+        "BEGIN",
+        "TRY",
+        Indent,
```

Comment:
In `BeginEndSegment` it's possible to have semicolons like:
```sql
BEGIN;
    SELECT 1;
END;
```
is it also possible to have semicolons after `BEGIN TRY`, `END TRY`, `BEGIN CATCH`, `END CATCH`?

Reply:
I just tested and it's possible to have semicolons after ```BEGIN TRY```, ```BEGIN CATCH```, and ```END CATCH```, but not after ```END TRY```.  I've added optional DelimiterSegments after ```BEGIN TRY``` and ```BEGIN CATCH```.  The one after ```END CATCH``` will be handled by the parent segment.

Reply:
perfect thanks for adding those